ui/app/src/state/reducers/preview.ts

```diff
@@ -537,7 +538,22 @@ const reducer = createReducer<GlobalState['preview']>(
     [setHighlightMode.type]: (state, { payload }) => ({
       ...state,
       highlightMode: payload.highlightMode
-    })
+    }),
+    [fetchGlobalPreferencesComplete.type]: (state, { payload }) => {
+      return {
+        ...state,
+        ...(nnou(payload.editMode) && { editMode: payload.editMode === 'true' }),
+        ...(nnou(payload.highlightMode) && { highlightMode: payload.highlightMode })
```

Comment:
Check for existence and set defaults. Should never be something other than the valid values for highlight mode.